### PR TITLE
Fix Ci failures related to array ragged input numpy deprecation

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

numpy 1.19 [deprecates](https://numpy.org/devdocs/release/1.19.0-notes.html#deprecate-automatic-dtype-object-for-ragged-input) (See [NEP 34](https://numpy.org/neps/nep-0034-infer-dtype-is-object.html)) automatic dtype for ragged input. This causes CI failures (please see #4731).

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by rfezzani** [comment]
CI failures are fixed by #4735

**Comment by sciunto** [comment]
I added the NEP in the top message.

File: skimage/transform/radon_transform.py [hunk 1]
```diff
@@ -75,7 +75,8 @@ def radon(image, theta=None, circle=True, *, preserve_range=False):
         shape_min = min(image.shape)
         radius = shape_min // 2
         img_shape = np.array(image.shape)
-        coords = np.array(np.ogrid[:image.shape[0], :image.shape[1]])
+        coords = np.array(np.ogrid[:image.shape[0], :image.shape[1]],
+                          dtype=object)
```
Comment (sciunto) [inline skimage/transform/radon_transform.py:79]: I might not be sufficiently awake, but I do not see why this is a ragged array.

**Comment by jni** [comment]
@meeseeksdev backport to 0.17.x

**Comment by jni** [comment]
Well, crap. 😂

**Pull request merged**